### DISCUSSION TITLE
Driver: Failed to open serial port: On One Installation Not the Other.

### DISCUSSION THREAD

**Discussion by PeteCal** [zwave-js/zwave-js-ui#2520] [Q&A]
Raspberry Pi 3B.  I installed zwavejs2mqtt on my RPi, once with one SD card and again with a second card.   The first one works fine, the second gives, "Driver: Failed to open serial port:Erroe no such file or directory, cannot open /dev/ttyACMO (ZW!00).  I am trying to ferret out what could be different.

The first card started with the basic imager, "Raspberry Pi OS......".   I then tried to install zwavejs2mqtt using snap but that failed.  So I used "Used https://flemmingss.com/how-to-set-up-zwavejs2mqtt-on-a-raspberry-pi-and-integrate-it-with-home-assistant/".  That worked and, once I straightened out my problems, My aeotec stick worked fine.  Note I was using this stick with another program so all the inclusions were completed.  I stopped before Home Assistant.

For the second card I started with the same imager, "Raspberry Pi OS...".  But I installed a number of python scripts that I will be using going forward. None of them involve zwave. Then I tried to install zwavejs2mqtt using snap as before.  That failed as before.  So I used https://flemmingss.com/how-to-set-up-zwavejs2mqtt-on-a-raspberry-pi-and-integrate-it-with-home-assistant/.  Everything seemed to go fine. but when I started 192.168.1.115:8091 I get the Driver: Failed....  

So I have been searching for why the first card works but the second doesn't.  I put the good card in and do screen captures of things like, lsusb, ls -al,  ls -l  /dev/tty*,  dmesg -T | grep ttyACM, dmesg, etc.  Then I swap out the "good" card and boot with the card that can't open the serial port.  I repeat the commands above and compare the output to the captured images.  So far I have not found a difference.  

Thinking settings were wrong I exported the settings from the working card and imported them into the failing card.   That didn't help anything.

Frankly I have run out of things to compare between th working version and the failing version.  I don't know where to go next.

Any ideas?

**Comment by robertsLando** [comment]
Try using /dev/serial-by/id path https://github.com/zwave-js/zwavejs2mqtt/discussions/2483

**Comment by PeteCal** [comment]
Thanks Robert
I will try that when I get a chance.  
I found a work around that let me continue developing.  
As I said I had a "basic, barebones" setup with Z2M and it worked.  
I had another SD card with all my applications and development environment.  Z2M wouldn't work because of the device problem.
So, I started with a copy of the card that had the working Z2M and began transfering my applications to it.  At each stage I checked that Z2M was still working.  I was able to get all my applications and Z2M all working on the new card.  I also am developing more applications and Z2M continues to work on the new card.

At some point I am sure I will have to reinstall the card with the non-functioning Z2M.  At that point I will follow your instructions and report back here.
Thanks for your help.
Pete
